src/pkg/ingress/bindings/binding_config.go

```diff
@@ -43,12 +43,39 @@ func (d *DrainParamParser) FetchBindings() ([]syslog.Binding, error) {
 			b.InternalTls = true
 		}
 
+		b.Type = getBindingType(urlParsed)
+
 		processed = append(processed, b)
 	}
 
 	return processed, nil
 }
 
+func getBindingType(u *url.URL) syslog.BindingType {
+	bindingType := syslog.BINDING_TYPE_LOG
+	switch u.Query().Get("drain-type") {
+	case "metrics":
+		bindingType = syslog.BINDING_TYPE_METRIC
+	case "all":
+		bindingType = syslog.BINDING_TYPE_ALL
+	case "allWithTimers":
```

Comment:
I find the parameter value confusing. Why is it `allowWithTimers` and it refers to aggregate drains? Why not simply `aggregate` as drain type?

Can you please explain what does allow with timers even mean?

Reply:
This is a very good question. 

the short answer of why it's allWithTimers is that...honestly I thought a work in progress name making a pr would get the idea across and I didn't want to wait forever sitting around trying to come up with something better. 

Right now "ALL" type doesn't actually allow all envelopes to pass through. So we have multiple types. All let's gauges and counters and logs through. Logs let's logs through, metrics lets gauges and counters through. However, as you may know, there exists more envelope types then that. 

The main one in use is https://github.com/cloudfoundry/loggregator-agent-release/blob/114b22af453b1b249cd1f7286532fe37b9c0327c/src/pkg/egress/syslog/rfc5424.go#L70 timers, which are in use for http start stop events, and I believe some of the newer cpu measures. There's been moves to try and push away from the usage of timers in general but it hasn't gotten that far. 

The other one is https://github.com/cloudfoundry/loggregator-agent-release/blob/114b22af453b1b249cd1f7286532fe37b9c0327c/src/pkg/egress/syslog/rfc5424.go#L74 events? I don't think they're so much used, but they exist.

Aggregate drains get them because a lot of the work towards getting aggregate drains to work was for log-cache. Log-cache needed everything, and so did some people's external integrations I presume, so they were allowed to get everything. App drains were I guess restricted to only get the "good" metrics. 

So that leaves us in a tough situation if we want to make things more consistant, because anyone using "all" is likely to be surprised if they suddenly start getting timers and events, but what do you call "more then all". all-with-timers is extremly verbose, but technecially accurate without breaking backwards compatibility. 

I'd be open and interested if you have ideas for alternate schemes tho.

Reply:
In general
1) we've seen some people ask for metrics only aggregate drains
and
2) it seems foolish to me to enforce the end of timers through inconsistant egress fiat rather then via reworking diego and router to use the new types first.

Reply:
(The enum AGGREGATE_DRAIN could/should probubly also be changed. Maybe should have done it. Not super comfy with "all-with-timers", but might be the best we have).

Reply:
@Benjamintf1 Thanks for the clarification. It's good that we are thinking tin he same direction :) I also understood `allWithTimers` as everything that comes with `all` + the timers.

I find this change as a must have, but I would do it other way:
Why don't we use the same parameters as it is done in the [Log Cache CLI's tail command](https://github.com/cloudfoundry/log-cache-cli/blame/main/README.md#L66-L67), two parameters `envelope-type` and `envelope-class`? This way we'll have the same API on both places.

The other option that came to my mind if we don't want to change the current `drain-type=all` behavior, is to add another `include` parameter and make a union of the current `all` and `timers` and/or `events`.

I'm personally for unification of the API and do the same what is done in the Log Cache CLI.

What do you think about this?

Reply:
Honestly, I've never seen the envelope class parameter to be paticularly compelling. It seems to exist because a design space to make it a option rather then a user need. 

IF anything, the more natural division here is metrics(counters and gauges), logs(logs, maybe events but those kinda don't exist), and timers(aka traces).

Reply:
I agree with you, but let's have `timers` as a new type and `all` will return all types, logs, metrics and timers.

Reply:
> the more natural division here is metrics(counters and gauges), logs(logs, maybe events but those kinda don't exist), and timers(aka traces).

I like this division! What if
1. Replace `allWithTimers` with a `timers` type, and made the `logs` type include events.
1. Then either introduce a breaking change or feature flag to switch `all` to really send everything.

(1) doesn't seem like a breaking change in my mind given the seeming nonexistence of events, and it allows for users to at least have some way to retrieve timers if they want them. It's less efficient for sure, but you could, in that world, have two different drains one for `logs`/`metrics`/`all`, and another for `timers`.

(2) would move us toward an `all` type that makes sense.

Reply:
Some of that could also be a separate change if we just remove `allWithTimers` from this PR.

Reply:
Removed allWithTimers.

Added new parameter "drain-data". Drain data operates under "new" logs/metrics/traces/all split vs "all/logs-and-metrics for app drains/logs/metrics". Events under logs.

Reply:
Arguably events should be under events, but loggregator isn't a good eventing system. Events should be deprecated and removed.